apps/api/src/profiles/profiles.service.ts

```diff
@@ -1,27 +1,171 @@
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { PrismaClientUnknownRequestError } from '@prisma/client/runtime';
 import { CreateProfileDto } from './dto/create-profile.dto';
+import { PrismaService } from '../prisma/prisma.service';
 import { UpdateProfileDto } from './dto/update-profile.dto';
-import { Profile } from './entities/profile.entity';
+
+interface Resp {
+  message: string;
+  data?: unknown;
+}
 
 @Injectable()
 export class ProfilesService {
-  create(createProfileDto: CreateProfileDto): Profile {
-    return { name: createProfileDto.name };
+  constructor(private prismaService: PrismaService) {}
+
+  // Response Handler
+  Success(resp: Resp) {
+    return {
+      Success: true,
+      message: resp.message,
+      data: resp.data,
+    };
   }
 
-  findAll(): Profile[] {
-    return [{ name: 'test' }];
+  // Method to CREATE a new profile
+  async create(createProfileDto: CreateProfileDto) {
+    const ReadResp = await this.read(createProfileDto.authid);
+    const EmailResp = await this.EmailRead(createProfileDto.email);
+    if (ReadResp.data != null) {
+      throw new HttpException(
+        {
+          success: false,
+          error: 'User Exists',
+        },
+        HttpStatus.CONFLICT
+      );
+    }
+    if (EmailResp != null && EmailResp.data != null) {
+      throw new HttpException(
+        {
+          success: false,
+          error: 'Email already in use',
+        },
+        HttpStatus.CONFLICT
+      );
+    }
+    const d = new Date(createProfileDto.dob);
```

Comment:
This is not a bad idea but issue is consider the dates are deeply nested or a new dev joins. 

It will result in more bugs. The better option is on dto. You can put custom dto or transformers provided by nestjs.

https://docs.nestjs.com/techniques/serialization